functions/package.json

```diff
@@ -7,13 +7,16 @@
     "lint": "tslint --project tsconfig.json",
     "prettier": "prettier --write './**/*.{js,json,md,scss,ts,html}'",
     "prettier:check": "prettier --check './**/*.{js,json,md,scss,ts,html}'",
-    "build": "tsc",
+    "build": "yarn generate && tsc",
     "serve": "yarn build && firebase emulators:start --only functions",
     "shell": "yarn build && firebase functions:shell",
     "start": "yarn shell",
     "test": "mocha -r ts-node/register src/test/**/*.spec.ts",
     "deploy": "firebase deploy --only functions",
-    "logs": "firebase functions:log"
+    "logs": "firebase functions:log",
+    "clean": "rimraf src/codegen/*",
+    "create-models-barrel": "node create-barrel.js src/codegen/model",
+    "generate": "yarn clean && openapi-generator generate -i openapi.yaml -g typescript-angular -o src/codegen -Dmodels --additional-properties=\"stringEnums=true\" && yarn create-models-barrel"
```

Comment:
this now uses the `typescript-angular` generator since didn't really find any more suitable one that wouldn't require ton of manual configuration... Not optimal, but since we are only including the models and not any services I guess it doesn't matter that much. Let's change it something better comes up